src/badge/types.js

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright (c) Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+import * as React from 'react';
+import { HIERARCHY, SHAPE, COLOR, PLACEMENT, ROLES } from './constants.js';
+import type { OverrideT } from '../helpers/overrides.js';
+
+export type HierarchyT = $Values<typeof HIERARCHY>;
+export type ShapeT = $Values<typeof SHAPE>;
+export type ColorT = $Values<typeof COLOR>;
+export type PlacementT = $Values<typeof PLACEMENT>;
+export type RoleT = $Values<typeof ROLES>;
+
+export type OffsetT = string;
+
+export type InlineBadgeOverridesT = {
+  Badge?: OverrideT,
+};
+
+export type BadgeOverridesT = {
+  Root?: OverrideT,
+  Positioner?: OverrideT,
+  Badge?: OverrideT,
+};
+
+export type InlineBadgePropsT = {
+  color?: ColorT,
+  shape?: ShapeT,
+  hierarchy?: HierarchyT,
+  children: ?React.Node,
+  overrides?: InlineBadgeOverridesT,
+  hidden?: boolean,
+};
+
+export type BadgePropsT = {
+  horizontalOffset?: string,
+  verticalOffset?: string,
+  placement?: PlacementT,
+  children: ?React.Node,
+  content: ?React.Node,
+  overrides?: BadgeOverridesT,
+} & InlineBadgePropsT;
```

Comment:
I recommend switching to exact types and spreading to union `InlineBadgePropsT` with `BadgePropsT`. I believe in the current state it will resolve the `overrides` property to `InlineBadgeOverridesT` which I assume is not what you wanted.

What's the practical difference between `children: ?React.Node` and `children?: React.Node`? I've only written the latter.

```suggestion
export type InlineBadgePropsT = {|
  color?: ColorT,
  shape?: ShapeT,
  hierarchy?: HierarchyT,
  children?: React.Node,
  overrides?: InlineBadgeOverridesT,
  hidden?: boolean,
|};

export type BadgePropsT = {|
  ...InlineBadgePropsT,
  horizontalOffset?: string,
  verticalOffset?: string,
  placement?: PlacementT,
  children?: React.Node,
  content?: React.Node,
  overrides?: BadgeOverridesT,
|};
```

Reply:
According to the Flow docs, it [returns the type of the first property that matches its name](https://flow.org/en/docs/types/intersections/#toc-intersections-of-object-types). (Definitely seems backwards from a javascript perspective). So, I put `InlineBadgePropsT` after the rest of `BadgePropsT` to avoid the issue of `overrides` resolving to that of `InlineBadgePropsT`.

That said I prefer your approach from a readability standpoint.

_That_ said all of this is mute because I was able to remove `InlineBadgePropsT` and related types now that I've removed `InlineBadge` per your recommendation!